db/system_keyspace.cc

```diff
@@ -2674,12 +2670,40 @@ future<service::topology> system_keyspace::load_topology_state() {
                     some_row.get_as<sstring>("global_topology_request"));
             ret.global_request.emplace(req);
         }
+    }
+
+    ret.features = co_await do_load_topology_features_state(std::move(rs));
+
+    co_return ret;
+}
+
+future<service::topology_features> system_keyspace::load_topology_features_state() {
+    auto rs = co_await execute_cql(
+        format("SELECT host_id, supported_features, enabled_features FROM system.{} WHERE key = '{}'", TOPOLOGY, TOPOLOGY));
+    assert(rs);
 
-        if (some_row.has("enabled_features")) {
-            ret.features.enabled_features = decode_features(deserialize_set_column(*topology(), some_row, "enabled_features"));
+    co_return co_await do_load_topology_features_state(std::move(rs));
+}
+
+future<service::topology_features> system_keyspace::do_load_topology_features_state(::shared_ptr<cql3::untyped_result_set> rs) {
```

Comment:
Looks like this doesn't have to return a `future`.
Also it could be made a `static` function

Reply:
Maybe instead of `do_load_` a better prefix would be `parse_` or something?
To make it more clear that this function is a pure computation